### ISSUE TITLE
Hi Prisma Team! Prisma Migrate just crashed. 

### ISSUE THREAD

**Issue by oguzhan2142** [prisma/prisma#14462]
Hi Prisma Team! Prisma Migrate just crashed. 

## Command

`db pull`

## Versions
      
| Name        | Version            |
|-------------|--------------------|
| Platform    | darwin             | 
| Node        | v16.15.1           | 
| Prisma CLI  | 4.1.0              | 
| Engine      | 8d8414deb360336e4698a65aa45a1fbaf1ce13d8| 

## Error
```
Error: [libs/datamodel/connectors/dml/src/model.rs:494:29] Hi there! We've been seeing this error in our error reporting backend, but cannot reproduce it in our own tests. The problem is that we have a primary key in the model `Product` that uses the column `CategoryId` which we for some reason don't have in our internal representation. If you see this, could you please file an issue to https://github.com/prisma/prisma so we can discuss about fixing this. -- Your friendly prisma developers. 
```

Oops, an unexpected error occured!
[libs/datamodel/connectors/dml/src/model.rs:494:29] Hi there! We've been seeing this error in our error reporting backend, but cannot reproduce it in our own tests. The problem is that we have a primary key in the model `Product` that uses the column `CategoryId` which we for some reason don't have in our internal representation. If you see this, could you please file an issue to https://github.com/prisma/prisma so we can discuss about fixing this. -- Your friendly prisma developers. 

<img width="1346" alt="Screen Shot 2022-07-23 at 18 07 57" src="https://user-images.githubusercontent.com/42682090/180610852-c106747d-0f40-42a7-90f0-db014982b2de.png">

**Comment by jkomyno** [comment]
Hi, I have tried reproducing this issue [here](https://github.com/jkomyno/reprod-14462/actions/runs/2760155558) but I haven't had luck so far. Would please share a SQL dump of your database, and your `schema.prisma` file? We understand you may not feel comfortable sharing this private info in public, so another option would be sending those files privately to `schemas@prisma.io`. We can also sign NDAs if necessary.

Moreover, could you please describe how the database was created and/or update? For instance, did you start from an existing SQL database that you introspected and maybe later modified via `prisma`?

**Comment by tomhoule** [comment]
This should be fixed in Prisma 4.2, to be released tomorrow (https://github.com/prisma/prisma-engines/pull/3108)
